src/gui/layout/qgslayoutlabelwidget.cpp

```diff
@@ -277,13 +277,29 @@ void QgsLayoutLabelWidget::fontChanged()
 {
   if ( mLabel )
   {
-    QFont newFont = mFontButton->currentFont();
     mLabel->beginCommand( tr( "Change Label Font" ), QgsLayoutItem::UndoLabelFont );
-    mLabel->setFont( newFont );
+    //disconnectUpdateSignal();
+    mLabel->setTextFormat( mFontButton->textFormat() );
+    //connectUpdateSignal();
```

Comment:
The connect disconnect signal are used elsewhere with this renderer, but in my test I didn't find them useful, hence why I commented them out in case they would be needed.